src/Tools/BuildValidator/Util.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace BuildValidator
+{
+    internal sealed record PortableExecutableInfo(string FilePath, Guid Mvid, bool IsReadyToRun);
+    internal static class Util
+    {
+        internal static PortableExecutableInfo? GetPortableExecutableInfo(string filePath)
+        {
+            using var stream = File.OpenRead(filePath);
+            var peReader = new PEReader(stream);
+            if (GetMvidForFile(peReader) is { } mvid)
+            {
+                var isReadyToRun = IsReadyToRunImage(peReader);
+                return new PortableExecutableInfo(filePath, mvid, isReadyToRun);
+            }
+
+            return null;
+        }
+
+        internal static Guid? GetMvidForFile(PEReader peReader)
```

Comment:
Feels like this could be a local function in GetPortableExecutableInfo

Reply:
I would prefer to leave it this way for now. These tools usually end up grabbing MVID as a separate operation which is why I kept it as a top level method here.